src/sentry/attachments/base.py

```diff
@@ -60,11 +68,11 @@ def from_upload(cls, file, **kwargs):
             name=file.name, content_type=file.content_type, data=file.read(), **kwargs
         )
 
-    @property
-    def data(self) -> bytes:
+    def load_data(self, project: Project | None = None) -> bytes:
         if self.stored_id:
-            # TODO: fetch the contents based on `stored_id`
-            raise NotImplementedError()
+            assert project
```

Comment:
What's the reasoning for passing this as `Project | None` and then asserting it?